syntax = "proto3";

service C2 {
    rpc StartJob(StartJobRequest) returns (StartJobResponse) {}
    rpc GetJob(GetJobRequest) returns (Job) {}
    rpc StopJob(StopJobRequest) returns (StopJobResponse) {}
}

message StartJobRequest {
    int32 commandId = 1;
    string userId = 2;
}

message StartJobResponse {
    enum Response {
        SUCCESS = 0;
        FAIL = 1;
    }
    Response response = 1;
    repeated Bot bots = 2;
}

message Bot {
    int32 botId = 1;
    enum State {
        ONLINE = 0;
        OFFLINE = 1;
    }
    State state = 2;
}

message GetJobRequest {
    string userId = 1;
}

message Job {
    int32 jobId = 1;
    int32 commandId = 2;
    string userId = 3;
    enum State {
        STARTING = 0;
        RUNNING = 1;
        FINISHED = 2;
    }
    State state = 4;
}

message StopJobRequest {
    string userId = 1;
}

message StopJobResponse {
    enum Response {
        SUCCESS = 0;
        FAIL = 1;
    }
    Response response = 1;
}

